File: Ch03. Basic Data Structures/Java code/StackAndQueueAsLinkedList/Queue.java

public class Queue {

	private Node head, tail;
	public Queue() {
		this.head = this.tail = null;
	}
	public boolean isEmpty() {
		return this.head == null;
	}
	public int size() {
		int counter = 0;
		for(Node tmp = this.head; tmp != null; tmp  = tmp.next)
			counter++;
		return counter;
	}
	public Integer dequeue() {
		if(this.isEmpty())
			return null;
		else {
			Integer i = this.head.info;
			if(this.head == this.tail)
				this.head = this.tail = null;
			else {
				this.head = this.head.next;
			}
			return i;
		}
	}
	public void enqueue(int i) {
		if(this.isEmpty())
			this.head = this.tail = new Node(i);
		else {
			this.tail.next = new Node(i);
			this.tail = this.tail.next;
		}
	}
}

File: Ch03. Basic Data Structures/StackAndQueueAsLinkedList/Queue.py
from Node import Node

class Queue:
	def __init__(self):
		self.head = self.tail = None

	def isEmpty(self):
		return self.head == None

	def size(self):
		tmp = self.head
		count = 0
		while tmp!= None:
			tmp = tmp.next
			count += 1
		return count

	def enqueue(self, payload):
		self.head = Node(payload, self.head)
		if self.tail == None:
			self.tail = self.head

	def dequeue(self):
		if self.isEmpty():
			return None
		else:
			payload  = self.tail.payload
			deleteMe = self.tail
			if self.head == self.tail:
				self.head = self.tail
			else:
				tmp = self.head
				while tmp.next != self.tail:
					tmp = tmp.next
				tmp.next = None
				self.tail = tmp
			deleteMe = None
			return payload